src/character.cpp

```diff
@@ -2645,3 +2647,58 @@ float Character::healing_rate( float at_rest_quality ) const
 
     return final_rate;
 }
+
+float Character::healing_rate_medicine( float at_rest_quality, const body_part bp ) const
+{
+    float rate_medicine = 0.0f;
+    float bandaged_rate = 0.0f;
+    float disinfected_rate = 0.0f;
+
+    if( has_effect( effect_bandaged, bp ) ) {
+        const effect &e_bandaged = get_effect( effect_bandaged, bp );
```

Comment:
`has_effect` gets effect on the inside, so you're getting the effect twice.
Instead you could call `get_effect` and then check if it was null by calling `e_bandaged.is_null()`.